tensorboard/embedding.py

```diff
@@ -1,26 +1,33 @@
 import os
 
+
 def make_tsv(metadata, save_path):
     metadata = [str(x) for x in metadata]
     with open(os.path.join(save_path, 'metadata.tsv'), 'w') as f:
         for x in metadata:
-            f.write(x+'\n')
+            f.write(x + '\n')
 
 
 # https://github.com/tensorflow/tensorboard/issues/44 image label will be squared
 def make_sprite(label_img, save_path):
+    # we have to make labels_img squared!
     import math
     import torch
     import torchvision
-    nrow = int(math.floor(math.sqrt(label_img.size(0))))
-    xx = torchvision.utils.make_grid(torch.Tensor(1,3,32,32), padding=0)
-    if xx.size(2)==33: # https://github.com/pytorch/vision/issues/206
+    # this ensure we have enought space for the images
+    nrow = int(math.ceil((label_img.size(0)) ** 0.5))
+    # cat the images to reach the square of base_size
+    label_img = torch.cat((label_img, torch.randn(nrow ** 2 - label_img.size(0), *label_img.size()[1:]) * 255), 0)
+    # this ensure no pixel are appended by make_grid call in save_image (such a stupid function)
```

Comment:
yep :) I suggest random instead of zeros, because if something goes wrong in the visualization, noise is more easily recognizable.